dify/weaviate_docs.py

```diff
@@ -1,103 +1,348 @@
+
+
+def process_markdown_file(
+    filepath: str,
+    docs_dir: str,
+    splitter: RecursiveCharacterTextSplitter,
+    batch: Any,
+    ollama_url: str = DEFAULT_OLLAMA_EMBED_URL,
+    model: str = DEFAULT_OLLAMA_MODEL,
+) -> bool:
+    """
+    Process a single markdown file and add it to the batch.
+
+    :param filepath: full path to the markdown file
+    :param docs_dir: base documentation directory
+    :param splitter: text splitter instance
+    :param batch: Weaviate batch object
+    :param ollama_url: Ollama API endpoint URL
+    :param model: model name for embeddings
+    :return: True if successful, False if failed
+    """
+    filename = os.path.basename(filepath)
 
-with collection.batch.fixed_size(BATCH_SIZE) as batch:
-    for root, _, files in os.walk(DOCS_DIR):
-        for filename in files:
-            if not any(filename.endswith(ext) for ext in ALLOWED_EXTENSIONS):
+    try:
+        _LOG.debug("Processing file: %s", filename)
+
+        # Load and split the document.
+        loader = UnstructuredMarkdownLoader(filepath)
+        docs = loader.load()
+        chunks = splitter.split_documents(docs)
+
+        chunk_count = 0
+        # Process each chunk.
+        for chunk in chunks:
+            text = chunk.page_content.strip()
+
+            # Skip empty chunks.
+            if not text:
                 continue
-            filepath = os.path.join(root, filename)
-
-            try:
-                loader = UnstructuredMarkdownLoader(filepath)
-                docs = loader.load()
-                chunks = splitter.split_documents(docs)
-
-                for chunk in chunks:
-                    text = chunk.page_content.strip()
-
-                    # Call Ollama embed API
-                    response = requests.post(
-                        OLLAMA_EMBED_URL,
-                        headers={"Content-Type": "application/json"},
-                        json={"model": OLLAMA_MODEL, "input": text},
-                    )
-                    if response.status_code != 200:
-                        raise RuntimeError(
-                            f"Ollama error {response.status_code}: {response.text}"
-                        )
-
-                    embedding_list = response.json().get("embeddings")
-                    if not embedding_list or not isinstance(
-                        embedding_list[0], list
+
+            # Get embedding from Ollama.
+            embedding = get_ollama_embedding(text, ollama_url, model)
+
+            # Add to batch.
+            batch.add_object(
+                properties={
+                    "text": text,
+                    "filename": filename,
+                    "filepath": os.path.relpath(filepath, docs_dir),
+                },
+                vector=embedding,
+            )
+            chunk_count += 1
+
+        _LOG.info("Successfully processed %s (%s chunks)", filename, chunk_count)
+        return True
+
+    except Exception as e:
```

Comment:
I'd prefer to make it fail. Otherwise, it completes and there were problems. Always better to break.